pulumi/folio/index.ts

```diff
@@ -0,0 +1,203 @@
+        fromPort: 0,
+        toPort: 0,
+        protocol: "-1",
+        cidrBlocks: ["0.0.0.0/0"]
+    }]
+});
+
+// Create our own IAM role and profile which we can bind nto the cluster's
+// NodeGroup. Cluster will also create a default for us, but we show here how
+// to create and bind our own.
+const managedPolicyArns: string[] = [
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+];
+
+// Creates a role and attaches the EKS worker node IAM managed policies.
+export function createRole(name: string): aws.iam.Role {
+    const role = new aws.iam.Role(name, {
+        tags: { "Name": name, ...tags },
+
+        assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+            Service: "ec2.amazonaws.com",
+        }),
+    });
+
+    let counter = 0;
+    for (const policy of managedPolicyArns) {
+        // Create RolePolicyAttachment without returning it.
+        const rpa = new aws.iam.RolePolicyAttachment(`${name}-policy-${counter++}`,
+            { policyArn: policy, role: role },
+        );
+    }
+
+    return role;
+}
+
+// We could create multiple roles and instance profiles here if we wanted to
+// perhaps to apply different profiles to different node groups.
+const role = createRole("folio-worker-role");
+const instanceProfile = new aws.iam.InstanceProfile("folio-instance-profile", { role: role });
+
+// Create an EKS cluster.
+const clusterName = "folio-cluster";
+const cluster = new eks.Cluster(clusterName, {
+    // Applies tags to all resources under management of the cluster.
+    tags: { "Name": clusterName, ...tags },
+
+    // Apply tags to this cluster.
+    clusterTags: { "Name": clusterName, ...tags },
+
+    // Tell the cluster about its VPC.
+    vpcId: vpc.id,
+    publicSubnetIds: vpc.publicSubnetIds,
+    privateSubnetIds: vpc.privateSubnetIds,
+    nodeAssociatePublicIpAddress: false,
+
+    // Need this because we're going to create our own NodeGroup below.
+    skipDefaultNodeGroup: true,
+
+    // Bind the SecurityGroup to the cluster. If we don't do this it will create
+    // its own default one.
+    clusterSecurityGroup: sg,
+
+    // Bind any roles we have created to the cluster.
+    instanceRoles: [role],
+
+    // Enable control plane logging. This sends logs to CloudWatch.
+    enabledClusterLogTypes: [
+        "api",
+        "audit",
+        "authenticator",
+    ],
+});
+
+// Create the node group with a bit more control than we would be given with the
+// defaults. Using this approach we could create multiple node groups if we wanted to
+// each with its own properties and InstanceProfile.
+const nodeGroupName = "folio-node-group";
+cluster.createNodeGroup(nodeGroupName, {
+    instanceType: aws.ec2.InstanceType.T3_XLarge,
+    desiredCapacity: 4,
+    minSize: 3,
+    maxSize: 5,
+
+    // TODO What is this doing?
+    labels: { "ondemand": "true" },
```

Comment:
I think this lets us know that these are the more expensive `on-demand` instances rather than `spot` instances